lib/telephony.dart

```diff
@@ -632,7 +632,7 @@ class SmsMessage {
           this.threadId = int.tryParse(value);
           break;
         case _SmsProjections.TYPE:
-          this.type = SmsType.values[value];
+          this.type = SmsType.values[int.parse(value)];
```

Comment:
I think we should still use `tryParse`. If the value is incorrect it will still throw an error. But just using `parse` will throw an error even if parsing fails for some reason.

Reply:
Ok, in fact you used `tryParse()` also for DATE and TIMESTAMP, leaving the SmsMessage field to null if the parsing fails